docs/30DaysOfPWA/advanced-capabilities/06.md

```diff
@@ -1,15 +1,170 @@
+
+It's a really good idea to let the user decide if they want to use this feature or not. Using `PeriodicSyncManager` interface of the registered service worker you can list existing "periodic syncs" to set the value (on/off) of UI controls (switches, checkboxes, etc) labeled like "Update feed in the background". When the user turns on this feature, you should register a new periodic sync:
+
+```javascript
+async function registerPeriodicFeedUpdate() {
+  const registration = await navigator.serviceWorker.ready;
+
+  // Query and check permission. See "Privacy and resource utilization considerations" section below.
+  const status = await navigator.permissions.query({
+    name: 'periodic-background-sync',
+  });
+  if (status.state !== 'granted') {
+    console.log('Periodic Background Sync is not granted.');
+    return;
+  }
+
+  try {
+    await registration.periodicSync.register('update-feed-content', {
+      minInterval: 24 * 60 * 60 * 1000, // We ask browser to run the sync no more than once a day
+    });
+    showNotification('Success! Feed will be updated in the background.');
+
+  } catch() {
+    console.error('Periodic Background Sync registration failed');
+  }
+}
+```
+
+Like always in PWA, feature detection is a best practice:
+
+```javascript
+if ('serviceWorker' in navigator && 'PeriodicSyncManager' in window) {
+  registerPeriodicFeedUpdate(post);
+} else {
+  console.log('Periodic Background Sync is not supported');
+}
+```
+
+In the service worker, you should listen and react to the `periodicsync` event named `update-feed-content`:
+
+```javascript
+self.addEventListener('periodicsync', event => {
+  if (event.tag === 'update-feed-content') {
+      event.waitUntil(
+
+        // You have to implement the function that fetches the latest posts and updates the storage:
+        updateFeedContent();
+      );
+  }
+});
+```
+
+Please note, that it's not an obligation but a possibility for you to fetch new data in `updateFeedContent` function. Before sending a server request, you might want to check if the user is on a non-metered connection and if there is enough space in the browser storage using [Network Information](https://developer.mozilla.org/en-US/docs/Web/API/Network_Information_API) and [StorageManager](https://developer.mozilla.org/en-US/docs/Web/API/Storage_API) APIs respectively.
+
+## Privacy and resource utilization considerations
+
+Both Background Sync and Periodic Background Sync APIs are designed to run some custom code (that might send network requests) in the background, at any time, and without notifying a user. It raises at least two concerns: privacy and resource utilization. Any network request is potentially a history leaking and location tracking reason. And any code consumes CPU, memory, battery, and, if needed, network resources. To avoid these problems, both specifications have requirements and recommendations that browser vendors must follow while implementing these APIs. They include:
```

Comment:
```suggestion
Both the Background Sync and Periodic Background Sync APIs are designed to run some custom code (that might send network requests) in the background, at any time, without notifying a user. It raises at least two concerns: privacy and resource utilization. Any network request is potentially a history-leaking and location-tracking opportunity. And any code execution consumes CPU, memory, battery, and, if needed, network resources. To avoid these problems, both specifications require:
```